### PR TITLE
Add cache command to sky_tools with subcommands for populating/clearing

### PR DESCRIPTION
**Pull request by jamesr**
This adds the following commands to sky_tools:
  sky_tools cache clear: Nukes all local artifacts in the cache
  sky_tools cache populate: Populates the cache with all known artifacts

This is useful both to fix busted caches and to make sure that the cache is
fully populated so that subsequent operations can proceed without needing
network access.

### CODE REVIEW

File: lib/src/cache.dart [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    ArtifactStore artifacts = new ArtifactStore(packageRoot);
+    await artifacts.populate();
+    return 0;
+  }
+
+  @override
+  Future<int> processArgResults(ArgResults results) async {
+    if (results['help'] || results.command == null) {
+      print(parser.usage);
```
Comment (jamesr) [inline lib/src/cache.dart]: This doesn't print anything useful about subcommands, even though the 'parser' command knows about them.  Is there a way to convince it to print that out or do I need to format them myself? (if the latter, I think the thing to do is extract this stuff from bin/sky_tools.dart).

Comment (abarth) [inline lib/src/cache.dart]: I'm not sure how to do that.

**Comment by abarth** [comment]
LGTM

File: lib/src/cache.dart (hunk 1 above)
Comment (jamesr) [inline lib/src/cache.dart]: Hmm, ok.  I'll land this and refactor the thing sky_tools.dart does to be more easily shared with this in a follow up.

File: lib/src/cache.dart [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      return 0;
+    }
+    if (results.command.name == 'clear') {
+      return _clear(results['package-root'], results.command);
+    } else if (results.command.name == 'populate') {
+      return _populate(results['package-root'], results.command);
+    }
+  }
```
Comment (jamesr) [inline lib/src/cache.dart]: Ah, the analyzer seems upset that this doesn't end with a return statement.

**Pull request merged**